src/views/datasets/hospitalization-hotspot/dashboard.vue

```diff
@@ -0,0 +1,347 @@
+
+const clusterIdToCluster = (id) => {
+  if (id === NULL_CLUSTER.cluster_id) {
+    return NULL_CLUSTER;
+  } else {
+    const { name, cluster_id } =
+      data.stats.find((s) => s.cluster_id === id) ?? NULL_CLUSTER;
+    return { name, cluster_id };
+  }
+};
+
+const activeCluster = ref(NULL_CLUSTER);
+useQueryParam({
+  param: "cluster",
+  ref: activeCluster,
+  valid: (id) =>
+    id === NULL_CLUSTER.cluster_id ||
+    data.stats.some((s) => s.cluster_id === id),
+  convertInt: true,
+  paramToVal: clusterIdToCluster,
+  valToParam: (cluster) => cluster.cluster_id,
+});
+const dashboardActiveCluster = ref(activeCluster.value);
+
+const dateControls = ref(props.currentDate ?? props.dates[0]);
+
+const emit = defineEmits(["newDate"]);
+watch(
+  () => dateControls,
+  (item) => {
+    emit("newDate", item.value);
+  },
+  { deep: true }
+);
+
+const updateDate = (newDate) => {
+  dateControls.value = newDate;
+};
+
+const townDefault = "All of Rhode Island";
+const towns = RI_GEOJSON.map((geo) => geo.properties.name).sort();
+const hez = HEZ_GEOJSON.map((geo) => geo.properties.HEZ).sort();
+
+const dropDowns = {
+  town: {
+    icon: "fas fa-globe",
+    label: "Where do you want to look?",
+    values: [townDefault, "--Health Equity Zone", ...hez, "--Towns", ...towns],
+  },
+  focusStat: {
+    label: "Which group do you want to focus on?",
+    icon: "fas fa-fill-drip",
+    values: [
+      { name: "All Residents", value: "total" },
+      { name: "White Residents", value: "white" },
+      { name: "Black Residents", value: "black" },
+      { name: "Latino Residents", value: "latino" },
+      { name: "Asian Residents", value: "asian" },
+    ],
+  },
+};
+
+const controls = ref({
+  focusStat: dropDowns.focusStat.values[0],
+  town: townDefault,
+});
+useQueryParam({
+  param: "stat",
+  ref: controls,
+  refField: "focusStat",
+  valid: (p) => dropDowns.focusStat.values.some((v) => p === v.value),
+  valToParam: (v) => v.value,
+  paramToVal: (p) => dropDowns.focusStat.values.find((v) => p === v.value),
+});
+useQueryParam({
+  param: "town",
+  ref: controls,
+  refField: "town",
+  valid: (p) => dropDowns.town.values.some((v) => p === v),
+  resetFields: ["cluster", "zoom"],
+});
+
+const updateControls = (newControls) => {
+  // zoom back out, selections don't make sense with zoom in
+  if (zoomed.value) {
+    zoomed.value = false;
+  }
+
+  // new town selected, unselect active cluster
+  if (newControls.town !== controls.value.town) {
+    // activeCluster gets reset through useQueryParam, therefore we just need to update dashboardActiveCluster
+    dashboardActiveCluster.value = NULL_CLUSTER;
+  }
+
+  // update the control selections
+  for (const [k, v] of Object.entries(newControls)) {
+    controls.value[k] = v;
+  }
+};
+
+const updateCluster = (newClusterId) => {
+  activeCluster.value = clusterIdToCluster(newClusterId);
+  dashboardActiveCluster.value = activeCluster.value;
+};
+</script>
+
+<style scoped>
```

Comment:
We could probably do this in another issue you want to, but we can move a lot of these common styles into the `src/assets/styles/main.scss` file